app/code/Magento/InventoryInStorePickupAdminUi/Plugin/Ui/DataProvider/ConvertBooleanToStringPlugin.php

```diff
@@ -48,12 +48,14 @@ public function afterGetData(
      *
      * @return array
      */
-    private function convertExtensionAttributeBooleanToIntStr(array $item):array {
-        if (isset($item[InStorePickupInterface::EXTENSION_ATTRIBUTES_KEY]) &&
-            isset($item[InStorePickupInterface::EXTENSION_ATTRIBUTES_KEY][InStorePickupInterface::IN_STORE_PICKUP_CODE])
-        ) {
-            $item[InStorePickupInterface::EXTENSION_ATTRIBUTES_KEY][InStorePickupInterface::IN_STORE_PICKUP_CODE] =
-                (string)(int)$item[InStorePickupInterface::EXTENSION_ATTRIBUTES_KEY][InStorePickupInterface::IN_STORE_PICKUP_CODE];
+    private function convertBooleanToString(array $item):array
+    {
+        if (isset($item[ExtensibleDataInterface::EXTENSION_ATTRIBUTES_KEY])) {
+            foreach ($item[ExtensibleDataInterface::EXTENSION_ATTRIBUTES_KEY] as $code => $value) {
+                if (is_bool($value)) {
+                    $item[ExtensibleDataInterface::EXTENSION_ATTRIBUTES_KEY][$code] = (string)(int)$value;
```

Comment:
Is there any way we can avoid double casting and so much nesting? Such coding fashion creates challenge to support in the future.

Reply:
A decided that this is something that need to be fixed in core.
So, I have created corresponding issue in core https://github.com/magento/magento2/issues/22067 and marked this plugin as deprecated.
As result, we no need to support in the future and can remove when it will be fixed in core.

Reply:
There is no need to deprecate plugins, as they are not supposed to be used as a contract anyway. Plugins are not covered by general Backwards Compatibility rules and usually are not marked with `api` or `deprecated`. Nevertheless, we have examples of `api` plugins in the code, but let's consider it as an exceptional case.

Reply:
I want to show that this is not something that other should use as example.
Let do not make exceptions, I have replaced `@deprecated` with `@TODO`.